code/clue.js

```diff
@@ -1,59 +1,363 @@
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Mystery object
+let mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+};
 
-// This function will be invoked when you click on the killer card.
+//This shuffles the favourite weapon
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+  const randomWeapon = randomSelector(weapons);
+  suspect.favouriteWeapon = randomWeapon.id;
+  });
+};
+
+
+// Pick killer function
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      shuffleFavouriteWeapon(suspects); //This randomly selects a favourite weapon.
+      mystery.killer = randomSelector(suspects);
+      document.getElementById('killerCard').style.background = mystery.killer.color;
+      document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById('killerAge').innerHTML = mystery.killer.age;
+      document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation;
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+      document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+      killerIsPicked = true;
+      }, 2500);
+    }
+};
+
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+// Pick weapon function
+const pickWeapon = () => {
+  if (!weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible';
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden';
+      mystery.weapon = randomSelector(weapons); 
+      document.getElementById('weaponImage').src = mystery.weapon.image;
+      document.getElementById('weaponName').innerHTML = mystery.weapon.name;
+      document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}kg`;
+      weaponIsPicked = true;
+    }, 2500)
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
```

Comment:
You don't need this line since you are using "onclick" in the HTML-file. Either you keep this in the JS-file and remove the "onclick" or vice versa.

Please update you PR and I will later on approve it!

Reply:
Thanks - well spotted. I have removed the eventListener